client/gutenberg/extensions/shared/atavist/frontend-management.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Wordpress dependencies
+ */
+
+import { createElement, render } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+
+export class FrontendManagement {
+
+	blockIterator( rootNode, blocks ) {
+		blocks.forEach( block => {
+			this.initializeFrontendReactBlocks(
+				block.component,
+				block.options,
+				rootNode
+			)
+		} )
+	}
+
+	initializeFrontendReactBlocks( component, options = {}, rootNode ) {
+		const { name, attributes } = options.config;
+		const { selector } = options;
+		const blockClass = [ '.wp-block', name.replace('/', '-') ].join( '-' );
+		rootNode
+			.querySelectorAll( blockClass )
+			.forEach( node => {
+				const data = this.extractAttributesFromContainer( node.dataset, attributes );
+				const children = this.extractChildrenFromContainer( node );
+				const el = createElement( component, data, children );
+				render( el, selector ? node.querySelector( selector ) : node );
+			} )
+	}
+
+	extractAttributesFromContainer( dataset, attributes ) {
+		const data = {};
+		for ( const name in attributes ) {
+			const attribute = attributes[ name ];
+			data[ name ] = dataset[ name ];
+			if ( attribute.type === 'array' || attribute.type === 'object' ) {
+
+				try {
+					data[ name ] = JSON.parse( data[ name ] );
+				} catch(e) {
+					// console.log( 'Error decoding JSON data for field ' + name, e);
+					data[ name ] = null;
+				}
+
+			}
+		}
+		return data;
+	}
+
+	extractChildrenFromContainer( node ) {
+    	const children = [];
```

Comment:
There seem to be indentation issues here.

Reply:
🤔At first glance, I'm not seeing any—this file looks to be all tabbed out. Specifically what did you notice?

Reply:
line 60 has a tab but line 61 has spaces

Reply:
Right you are! Fixed in https://github.com/Automattic/wp-calypso/pull/27836/commits/1929d54755a21d7ff44905f66f55fb800ba2723d. How did you spot these?